catch errors in zen_workaround

GitMensch opened the pull request:
and ensure handle is closed

side issue of #3251

rocallahan commented on scripts/zen_workaround.py:
```diff
@@ -53,11 +53,26 @@ def is_secure_boot_enabled():
+    except OSError as e:
+        sys.stderr.write(str(e) + '\n')
+        print("General error on reading from MSR.")
+        return 0
```
Why is returning 0 a good idea? Shouldn't we kill the script?

GitMensch replied:
Returning 0 runs into "not supported" later on and is similar what was done in the other place.
If you want to I can drop the return 0 and break / continue which we have into killing the script after the error (which also may be placed into stderr completely).

GitMensch commented:
OK, I'll adjust both the old and new code to output to stderr and exit directly.

GitMensch commented:
Adjusted to exit the script as early as possible when errors occur; added a note to the secure-boot branch as we now never reach the end of the script any more.

GitMensch commented:
@rocallahan your requested changes were done a year ago, are there more things you want to have adjusted or should this be merged / just closed?

rocallahan commented:
Merged, sorry for losing this
